fw/src/mppt/global/particleswarm.hpp

```diff
@@ -0,0 +1,395 @@
+/**
+ * @file particleswarm.hpp
+ * @author Afnan Mir (afnanmir@utexas.edu) & Nash Wu
+ * @brief Implementation of Particle Swarm Optimization GlobalMPPTAlgortihm.
+ * @version 0.2.0
+ * @date 2024-01-27
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include <random>
+
+#pragma once
+#include "../mppt.hpp"
+#include "../local/pando.hpp"
+
+#define    NUM_AGENTS 4
+#define    C1 0.02
+#define    C2 0.5
+#define    W 0.35
+
+class Particle final {
```

Comment:
you could probably define this class privately inside the ParticleSwarm algorithm class to keep it encapsulated.